src/gluonts/mx/block/dropout.py

```diff
@@ -0,0 +1,243 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+# Standard library imports
+from typing import Tuple
+
+# Third-party imports
+from mxnet.gluon.rnn import (
+    HybridRecurrentCell,
+    ModifierCell,
+    BidirectionalCell,
+    SequentialRNNCell,
+)
+
+# First-party imports
+from gluonts.core.component import validated
+from gluonts.model.common import Tensor
+
+
+class VariationalZoneoutCell(ModifierCell):
+    """
+    Applies Variational Zoneout on base cell.
+    Combine Zoneout with variational dropout, https://arxiv.org/pdf/1512.05287.pdf, 
+    "A Theoretically Grounded Application of Dropout in Recurrent Neural Networks".
+    Variational zoneout uses the same mask across time-steps. It can be applied to RNN
+    outputs, and states. The masks for them are not shared.
+
+    The mask is initialized when stepping forward for the first time and will remain
+    the same until .reset() is called. Thus, if using the cell and stepping manually without calling
+    .unroll(), the .reset() should be called after each sequence.
+
+    Parameters
+    ----------
+    base_cell : RecurrentCell
+        The cell on which to perform variational dropout.
+    zoneout_outputs : float, default 0.
+        The dropout rate for outputs. Won't apply dropout if it equals 0.
+    zoneout_states : float, default 0.
+        The dropout rate for state inputs on the first state channel.
+        Won't apply dropout if it equals 0.
+    
+    """
+
+    @validated()
+    def __init__(
+        self,
+        base_cell: HybridRecurrentCell,
```

Comment:
`HybridRecurrentCell` or `RecurrentCell`? (see docstring above)

Reply:
RecurrentCell is the superclass of HybridRecurrentCell, which is more general. Maybe RecurrentCell is better.